packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -16,14 +16,15 @@ import {
   FormControlLabel,
 } from "@mui/material";
 import { Typography } from "./Typography";
-import { memo, useCallback, useMemo, useState } from "react";
+import { memo, useCallback, useMemo } from "react";
 import { useTranslation } from "react-i18next";
 
 import type { SeleniumProps } from "./SeleniumProps";
+import { useControlledState } from "./useControlledState";
 
 export const checkboxValidityValues = ["valid", "invalid", "inherit"] as const;
 
-export type CheckboxProps = {
+type BaseCheckboxProps = {
```

Comment:
I'd like to find a better name for `BaseCheckboxProps` if possible. I get how it's used though.

Oh! I think we can do this differently.

Does this work?
```ts
export type CheckboxProps = (
  {
    name: string,
  }
  & (
    | {
      onChange: MuiCheckboxProps["onChange"]
      value: string
    }
    | {
      onChange?: MuiCheckboxProps["onChange"]
    }
  )
)
```